src/functional/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -337,6 +339,12 @@ def _collect_offset_or_axis_node(
             )
         return result
 
+    def _check_shift_scan(self, node: itir.FunCall) -> None:
+        if node.fun.id == "shift":
+            raise ValueError("unapplied shift call not supported: {node}")
+        elif node.fun.id == "scan":
+            raise ValueError("scans are only supported at the top level of a stencil closure")
+
```

Comment:
because I added the "cast" case, pre-commit was outputting that `visit_FunCall` is too complex (hence I had to make it simpler).

I am ok changing the name of `_check_shift_scan` to `_error_on_illegal_function_calls`